aws-cpp-sdk-core/source/utils/crypto/openssl/CryptoImpl.cpp

```diff
@@ -238,10 +256,7 @@ namespace Aws
             OpenSSLCipher::OpenSSLCipher(OpenSSLCipher&& toMove) : SymmetricCipher(std::move(toMove)),
                                                                    m_encDecInitialized(false)
             {
-                m_ctx = toMove.m_ctx;
-                toMove.m_ctx.cipher = nullptr;
-                toMove.m_ctx.cipher_data = nullptr;
-                toMove.m_ctx.engine = nullptr;
+                EVP_CIPHER_CTX_copy(m_ctx, toMove.m_ctx);
```

Comment:
This actually needs updating, as m_ctx hasn't been assigned yet. I believe a call to init() should be sufficient.